x-pack/legacy/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -91,25 +91,20 @@ export class WorkspaceHandler {
       await this.checkCommit(bareRepo, targetCommit);
       revision = defaultBranch;
     }
-    let workspaceRepo: Repository;
+    let workspace;
     if (await this.workspaceExists(bareRepo, repositoryUri, revision)) {
-      workspaceRepo = await this.updateWorkspace(repositoryUri, revision, targetCommit);
```

Comment:
@spacedragon also we are trying to avoid return the whole `workspaceRepo` Repository object and pass it around. this makes it hard to clean up resources.